lib/src/test/java/com/auth0/android/lock/AuthenticationCallbackTest.java

```diff
@@ -80,6 +82,16 @@ public void shouldReturnAuthentication() {
         assertThat(callback, hasNoError());
     }
 
+    @Test
+    public void shouldReturnAuthenticationError() {
+        Intent data = new Intent();
+        AuthenticationException error = new AuthenticationException("err_code", "err description");
+        data.putExtra(Constants.EXCEPTION_EXTRA, error);
+        callback.onEvent(LockEvent.AUTHENTICATION, data);
+
+        assertThat(callback, hasError());
```

Comment:
checks that when an "exception extra" is present, the `onError` is invoked with a non-null value for the exception